package.json

```diff
@@ -29,9 +33,6 @@
     "babel-core": "7.0.0-bridge.0",
     "babel-jest": "^23.6.0",
     "pug-lint-vue": "^0.1.3",
-    "stylus": "^0.54.5",
-    "stylus-loader": "^3.0.2",
-    "vue-cli-plugin-pug": "^1.0.7",
```

Comment:
For the sake of flexibility, I replaced this with pug and pug-plain-loader and move them from devDependencies to dependencies, since the consumer will need a loader to compile our pug. This allows them to use vue-cli-3, vue2, or whatever else with their own webpack configuration.

vue-cli already supports this out-of-the-box, so all you have to do is have pug installed.

https://github.com/vuejs/vue-cli/blob/f8e97838dd25e37433ebbdb92b748c48b89f7c2a/packages/%40vue/cli-service/lib/config/base.js#L131

Reply:
On #29, I intentionally replaced `pug-loader` and `pug-plain-loader` with `vue-cli-plugin-pug`. The rationale there is that the loader chain used by vue-cli-service to support pug is actually an internal detail, and using the official Vue plugin that supports pug in components means we don't have to know that implementation detail. While it may be obvious that it's pug and pug-loader, upstream vue-cli-service could add additonal loaders or loader configuration that requires other packages, and by virtue of using the plugin, we don't have to worry about that at all.

Regarding whether things belong in dependencies of devDependencies, I'm actually not sure what the best solution here, I'd like to hear @jbeezley 's thoughts on this as well. The way I had separated it, and the way @jbeezley chose to separate it on resonantgeo, was to put runtime dependencies in `dependencies` and build-time dependencies in `devDependencies`. It kind of depends in what context people are installing the package whether that makes sense -- if they are just installing it to use the prebuilt UMD module, they don't need anything other than perhaps a peer dependency on `vue`. If they want to build a downstream app using the source files in this package, they'll need all our runtime dependencies, but will likely be providing their own build toolchain, which I think is why we had it setup like that.

I'm not sure what the correct answer is, but I think it makes sense that all of our build time dependencies should be in the same place, and all of our runtime dependencies should be in the same place.

Reply:
Gotcha - sorry for the assumption here, and thanks for the thorough explanation.

I went through the exercise of creating a downsteam using this as source and these were the dependencies not already included by virtue of downstream necessarily being a Vue.js app.

Reply:
There really isn't a "correct" answer.  The philosophy of the webpack maintainers is that loaders and webpack configs are an application concern, so they should go in `devDependencies`, but you can come up with arguments for why that is bad.

I think since we are publishing a UMD bundle, we should optimize for it.  That means keeping them in `devDependencies` so that we minimize the packages that are installed.  If someone is importing the components directly, that is an advanced use case and they are on their own in terms of importing loaders.